contribs/gnodev/pkg/emitter/server_test.go

```diff
@@ -30,13 +31,15 @@ func TestServer_ServeHTTP(t *testing.T) {
 
 	u := "ws" + strings.TrimPrefix(s.URL, "http")
 	c, _, err := websocket.DefaultDialer.Dial(u, nil)
-	if err != nil {
-		t.Fatalf("client Dial failed: %v", err)
-	}
+	require.NoError(t, err, "client Dial failed")
+
 	defer c.Close()
 
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
```

Comment:
Avoid t.Parallel if you're using eventually, so that it is not executed together with other 1000 tests risking to miss the deadline